include/votca/csg/atomtable.h

```diff
@@ -0,0 +1,137 @@
+/* 
+ * Copyright 2009-2011 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef _atom_table_H
+#define _atom_table_H
+
+#include <vector>
+#include <string>
+
+namespace votca { namespace csg {
+
+class AtomTable{
+
+    /**
+     * Brief information about atom table
+     *
+     * The atom table class is responsible for stroring relevant information 
+     * from the periodic table.
+     *
+     */
+
+    public:
+        /**
+         * Constructor
+         */
+        AtomTable(void) {};
+        /* 
+         * Destructor 
+         */
+        ~AtomTable() {}
+
+        /* 
+         * Check to see if the symbol is a valid symbol. The symbols follow the
+         * standard seen in the periodic table, H - Hydrogen, He - Helium etc.
+         * Will return:
+         * true  - if symbol is real
+         * false - if symbol is not recognized
+         */
+        bool checkSymbol(std::string sym);
+        /*
+         * Will return the symbols associated with the halogen atoms in a vector.
+         */
+        std::vector<std::string> getHalogens(void);
+        /*
+         *  Will return the sybmols associated with the noble gasses 
+         */
+        std::vector<std::string> getNoble(void);
+
+        /* 
+         * Pass the symbol in and returns the mass in a.u.
+         * If the symbol is not recognized will exit with an error message
+         * If the mass is not known it will also exit with an error message
+         */
+        double getMass(std::string sym);
+        /*
+         * Pass the ymbol in and returns the atomic number.
+         * If the symbol is not recognized it will print an error message and exit.
+         */
+        int getAtomicNumber(std::string sym);
+
+    protected:
+
+        /*
+         * Return the index associated with the tables for the chosen element.
+         */
+        int getIndex(std::string sym);
+
+        /* 
+         * Stores the symbols for all the atoms in the periodic table.
```

Comment:
Instead of having vector of symbols, vector of masses and vector of atomic number, why not to have an unordered_map with the key being the symbol and the value being a struct with mass, atomic number
```c++
struct Atom {
  double mass;
  double atomicNumber;
};

std::unordered_map<std::string, Atom> atomMap;
atomMap.insert(std::make_pair("Ac", Atom{227.028, 89}));
...
```

You can of course in the constructor reuse the vectors that you already created and fill up the map in some loop.

With this you can reduce the complexity of all of the getters that you have.

Reply:
I like the idea of an unordered_map.